packages/cerebral/src/Module.js

```diff
@@ -1,35 +1,24 @@
 class Module {
-  constructor (path, module, controller) {
-    this.name = path.slice().pop()
-    this.path = path
-    this.controller = controller
-    this.modules = {}
+  constructor (controller, path, moduleDescription) {
+    const moduleStub = {
+      controller,
+      path: path.join('.'),
+      name: path.slice().pop()
+    }
 
-    const moduleDescription = typeof module === 'function' ? module(this) : module
+    const module = typeof moduleDescription === 'function' ? moduleDescription(moduleStub) : moduleDescription
```

Comment:
`moduleDescription(moduleStub)` returnes either moduleDescription like object or any class instance with special supported properties (state, signals, provider, modules).
Then we just return recieved object/instance from constructor to make "module" instance.
Just see a test and
https://github.com/cerebral/cerebral/blob/3e311f0/packages/cerebral-router/src/index.js#L33
https://github.com/cerebral/cerebral/blob/3e311f0/packages/cerebral-router/src/router.js#L5